MySQL reaches maximum number of connections

Package version (if known): v1.4.1

## Describe the bug

The build (test suite) is failing in master due to a failure in MySQL. Local tests have been done and:

- Test suite passes for PostgreSQL
- When removing a contrib, thus opening less amount of connections, the test passes for MySQL.
    - Tested ignoring `contrib_cern` and passes
    - Tested ignoring `contrib_keycloak` and passes (therefore, it is not a failure of a keycloak implementation)
- Run the specific test that fails (by itself) and passes

## Steps to Reproduce
1. Run the test suite on master

## Additional context

Stacktrace:
```
data = b'\xff\x10\x04#08004Too many connections'

    def raise_mysql_exception(data):
        errno = struct.unpack("<h", data[1:3])[0]
        errval = data[9:].decode("utf-8", "replace")
        errorclass = error_map.get(errno)
        if errorclass is None:
            errorclass = InternalError if errno < 1000 else OperationalError
>       raise errorclass(errno, errval)
E       sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (1040, 'Too many connections')
E       (Background on this error at: http://sqlalche.me/e/13/e3q8)

../../.local/share/virtualenvs/invenio-oauthclient-RJ1IXkwX/lib/python3.7/site-packages/pymysql/err.py:143: OperationalError
```

CC @max-moser who discovered the issue

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -159,6 +159,7 @@ def base_app(request):
             if str(db.engine.url) != 'sqlite://':
                 drop_database(str(db.engine.url))
             shutil.rmtree(instance_path)
+            db.engine.dispose()
 
     request.addfinalizer(teardown)
 

